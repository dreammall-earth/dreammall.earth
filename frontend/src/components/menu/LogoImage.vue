<template>
  <v-img class="w-100" :class="classes" :src="textEnabled ? Logo : LogoSimple" />
</template>

<script lang="ts" setup>
import { computed } from 'vue'

import LogoSimple from '#assets/dreammall-logo-simple.svg'
import Logo from '#assets/dreammall-logo.svg'

const props = withDefaults(
  defineProps<{
    /**
     * size of the logo image
     */
    size?: 'small' | 'medium' | 'large' | 'tiny'
    textEnabled?: boolean
  }>(),
  { size: 'medium', textEnabled: true },
)

const classes = computed(() => ({
  'logo-large': props.size === 'large',
  'logo-medium': props.size === 'medium',
  'logo-small': props.size === 'small',
  'logo-tiny': props.size === 'tiny',
}))
</script>

<style scoped lang="scss">
.logo-large {
  max-width: 36em;
}

.logo-medium {
  max-width: 14em;
}

.logo-small {
  max-width: 8em;
}

.logo-tiny {
  max-width: 2em;
}
</style>
