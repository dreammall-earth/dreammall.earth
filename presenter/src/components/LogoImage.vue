<template>
  <v-img class="w-100" :class="classes" :src="Logo" />
</template>

<script lang="ts" setup>
import { computed } from 'vue'

import Logo from '#assets/dreammall-logo.svg'

const props = withDefaults(
  defineProps<{
    /**
     * size of the logo image
     */
    size?: 'small' | 'medium' | 'large'
  }>(),
  { size: 'medium' },
)

const classes = computed(() => ({
  'logo-large': props.size === 'large',
  'logo-medium': props.size === 'medium',
  'logo-small': props.size === 'small',
}))
</script>

<style scoped lang="scss">
.logo-large {
  max-width: 36em;
}

.logo-medium {
  max-width: 14em;
}

.logo-small {
  max-width: 8em;
}
</style>
