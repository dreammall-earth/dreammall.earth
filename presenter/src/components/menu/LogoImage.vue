<template>
  <v-img class="ma-2 pa-1 w-100" :class="classes" :src="Logo" />
</template>

<script lang="ts" setup>
import Logo from '#assets/dreammall-logo.svg'
import { computed } from 'vue'

const props = withDefaults(
  defineProps<{
    /**
     * size of the logo image
     */
    size?: 'small' | 'medium' | 'large'
  }>(),
  { size: 'medium' },
)

const classes = computed(() => ({
  'logo--large': props.size === 'large',
  'logo--medium': props.size === 'medium',
  'logo--small': props.size === 'small',
}))
</script>

<style scoped lang="scss">
.logo--large {
  max-width: 36em;
}
.logo--medium {
  max-width: 14em;
}
.logo--small {
  max-width: 8em;
}
</style>