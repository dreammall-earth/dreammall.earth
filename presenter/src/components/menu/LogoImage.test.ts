import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'

import LogoImage from './LogoImage.vue'

describe('LogoImage', () => {
  let wrapper

  const Wrapper = () => {
    return mount(LogoImage)
  }

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders Logo', () => {
    expect(wrapper.find('.v-img').exists()).toBeTruthy()
    expect(wrapper.find('.v-img__img').attributes().src).toBe('/src/assets/dreammall-logo.svg')
  })

  describe('classes', () => {
    describe('default', () => {
      it('has class logo-medium', () => {
        expect(wrapper.find('.v-img').classes('logo-large')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-small')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-medium')).toBe(true)
      })
    })

    describe('size large', () => {
      beforeEach(async () => {
        await wrapper.setProps({
          size: 'large',
        })
      })

      it('has class logo-large', () => {
        expect(wrapper.find('.v-img').classes('logo-medium')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-small')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-large')).toBe(true)
      })
    })

    describe('size medium', () => {
      beforeEach(async () => {
        await wrapper.setProps({
          size: 'medium',
        })
      })

      it('has class logo-medium', () => {
        expect(wrapper.find('.v-img').classes('logo-large')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-small')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-medium')).toBe(true)
      })
    })

    describe('size small', () => {
      beforeEach(async () => {
        await wrapper.setProps({
          size: 'small',
        })
      })

      it('has class logo-small', () => {
        expect(wrapper.find('.v-img').classes('logo-large')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-medium')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-small')).toBe(true)
      })
    })

    describe('unexpected size', () => {
      beforeEach(async () => {
        await wrapper.setProps({
          size: 'unexpected',
        })
      })

      it('has no logo size class', () => {
        expect(wrapper.find('.v-img').classes('logo-large')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-medium')).toBe(false)
        expect(wrapper.find('.v-img').classes('logo-small')).toBe(false)
      })
    })
  })
})
