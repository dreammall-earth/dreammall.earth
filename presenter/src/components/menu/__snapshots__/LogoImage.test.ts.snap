// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogoImage > renders Logo 1`] = `
<div
  class="v-responsive v-img v-img--booting w-100 logo-medium"
  data-v-f6be2607=""
>
  <div
    class="v-responsive__sizer"
  />
  
  
  <transition-stub
    appear="true"
    css="true"
    name="fade-transition"
    persisted="false"
  >
    <img
      class="v-img__img v-img__img--contain"
      src="/src/assets/dreammall-logo.svg"
      style="display: none;"
    />
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="fade-transition"
    persisted="false"
  >
    <!---->
  </transition-stub>
  <!---->
  <!---->
  <!---->
  
  
  <!---->
</div>
`;
